src/components/virtual-scroll/virtual-scroll.ts

```diff
@@ -406,6 +405,28 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
   /**
    * @hidden
    */
+  firstRecord(): number {
+    const cells = this._cells;
+    if (cells.length > 0) {
+      return cells[0].record;
+    }
+    return 0;
+  }
+
+  /**
+   * @hidden
+   */
+  lastRecord(): number {
+    const cells = this._cells;
```

Comment:
LastRecord is modified with scrolling and reset only after needChanges is set to true at least once, so after scrolling all the way to the bottom all add/deletes/updates (except appends) will take slow path until a complete recalculation takes place, even if in the mean time I have scrolled all the way up to the bottom.
Not sure if there is a better approach on this, just pointing out the current state.